client/js/widgets/button.js

```diff
@@ -68,10 +68,28 @@ export class Button extends Widget {
       if(a.applyVariables) {
         if(Array.isArray(a.applyVariables)) {
           for(const v of a.applyVariables) {
-            if(v.parameter && v.variable)
+            if(v.parameter && v.variable) {
               a[v.parameter] = variables[v.variable];
-            else
+            }
+            else if (v.parameter && v.template) {
+              for (let key in variables) {
+                a[v.parameter] = v.template.replace(/({([^}]+)})/g, function(i) {
+                  let key = i.replace(/{/, '').replace(/}/, '');
+                  if (!variables[key]) return "";
+                  return variables[key];
+                });
+              }
+            }
+            else if (v.parameter && v.property) {
+              var targetWidget = (v.property.split(".")[0] === "this") ? widgets.get(this.id) : widgets.get(v.property.split(".")[0]);
+              if (typeof targetWidget != undefined) {
```

Comment:
I'm pretty sure this will never be false because the `typeof` returns a String. Should be `typeof targetWidget != "undefined"` or `targetWidget !== undefined`.